pkg/monitortests/kubeapiserver/legacykubeapiservermonitortests/early_e2e_disruption.go

```diff
@@ -0,0 +1,53 @@
+package legacykubeapiservermonitortests
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+// amount of time after E2E tests start that we consider "early"
+const earlyE2ECheckDuration = 3 * time.Minute
+
+func testEarlyE2EAPIServerDisruption(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "record api server disruption that occurs early in E2E tests"
+
+	// regex to match the backends we care about
+	var earlyE2EBackendRe = regexp.MustCompile(`^(cache-)?(openshift|kube|oauth)-api-`)
+
+	var e2eStart time.Time
+	var tests []*junitapi.JUnitTestCase
+	for _, event := range events {
+		switch event.Source {
+		case monitorapi.SourceE2ETest:
+			if e2eStart.IsZero() {
+				e2eStart = event.From
+			}
+		case monitorapi.SourceDisruption:
+			if event.Message.Reason == monitorapi.DisruptionBeganEventReason &&
+				event.From.Sub(e2eStart) < earlyE2ECheckDuration &&
```

Comment:
Is there a chance that disruption would begin before we saw our first e2e test and e2eStart would be 0?  Wondering if we need one pass to find the first e2eStart then second for disruption.  Or something..

Reply:
if e2eStart is 0 then the duration before the event will be much larger than `earlyE2ECheckDuration`

Reply:
it does feel odd to depend on the zero value perhaps. could check it explicitly as part of the expression here if that would be clearer...

Reply:
Ok, I'm reading that as event.From - 0 == some big # that is !< earlyE2ECheckDuration.

Didn't know if you wanted to check disruption within 3 minutes of the start or only after the start.

Reply:
That's a good idea, we should probably detect pre-existing disruptionbegan, which could be missed here if these are sorted. Probably requires two loops through the intervals, but the one scanning for first e2e test can be short circuited.